### PR TITLE
Add db sweeper extension and specs

### PR DESCRIPTION
**Pull request by StevenLeighton21**
Extend the DB sweeper call to remove SavedForm records older than 60 days
Extend the DB sweeper call to invalidate on SavedForm records older than 28 days
Implement invalidate function to blank sensitive data in records, called by sweeper

### CODE REVIEW

File: spec/models/saved_form_spec.rb [hunk 1]
```diff
@@ -2,10 +2,63 @@
...
 RSpec.describe SavedForm, type: :model do
   it { should validate_presence_of(:service_slug) }
-  it { should validate_presence_of(:user_id) }
 
   it 'should set the default values' do
     expect(subject.attempts).to eq(0)
     expect(subject.active).to eq(true)
   end
+
+  context '#invalidated' do
```
Comment (njseeto) [inline spec/models/saved_form_spec.rb:11]:
```suggestion
  describe '#invalidated' do
```

To keep it similar to the other specs

File: spec/models/saved_form_spec.rb [hunk 2]
```diff
@@ -2,10 +2,63 @@
...
+    let(:model) { SavedForm.new(user_id: '123', user_token: '456', user_data_payload: 'q1 => answer', secret_answer: 'hello', active: true, service_slug: 'some_slug') }
+    
+    it 'should consider the model invalidated if user id is empty' do
```
Comment (njseeto) [inline spec/models/saved_form_spec.rb:14]:
```suggestion
    context 'if user id is empty' do
      it 'should consider the model invalidated' do
        # assertion
      end
    end
```

This is mainly for readability and keeping it similar to the other spec

File: spec/models/saved_form_spec.rb [hunk 3]
```diff
@@ -2,10 +2,63 @@
...
+      expect(model.invalidated?).to be(false)
+      model.user_id = nil
+      expect(model.invalidated?).to be(true)
+      model.user_id = ""
+      expect(model.invalidated?).to be(true)
+    end
+
+    it 'should consider the model invalidated if user token is empty' do
```
Comment (njseeto) [inline spec/models/saved_form_spec.rb:22]:
```suggestion
    context 'if user token is empty' do
      it 'should consider the model invalidated' do
       end
     end
```
similar here

File: spec/models/saved_form_spec.rb [hunk 4]
```diff
@@ -2,10 +2,63 @@
...
+      expect(model.invalidated?).to be(false)
+      model.user_token = nil
+      expect(model.invalidated?).to be(true)
+      model.user_token = ""
+      expect(model.invalidated?).to be(true)
+    end
+
+    it 'should consider the model invalidated if attempts >3' do
```
Comment (njseeto) [inline spec/models/saved_form_spec.rb:30]:
```suggestion
    context 'if attempts >3' do
       it 'should consider the model invalidated' do
       end
     end
```

File: spec/models/saved_form_spec.rb [hunk 5]
```diff
@@ -2,10 +2,63 @@
...
+      expect(model.invalidated?).to be(false)
+      model.increment_attempts!
+      expect(model.invalidated?).to be(false)
+      model.increment_attempts!
+      expect(model.invalidated?).to be(false)
+      model.increment_attempts!
+      expect(model.invalidated?).to be(true)
+    end
+
+    it 'should consider the model invalidated if marked inactive' do
```
Comment (njseeto) [inline spec/models/saved_form_spec.rb:42]:
```suggestion
    context 'if marked inactive' do
      it 'should consider the model invalidated' do
      end
    end
```

**Review by njseeto — COMMENTED** [review]
looks good, couple of small suggestions

File: app/services/db_sweeper.rb [hunk 6]
```diff
@@ -1,11 +1,23 @@
 class DbSweeper
   def call
     UserData.where("created_at < ?", age_threshold).destroy_all
+    SavedForm.where("created_at < ?", invalidation_threshold).each do |record|
```
[the remark this answers is not in the document]
Comment (StevenLeighton21) [reply app/services/db_sweeper.rb:4]: I wanted to keep them separate - what if we want to have different age thresholds? Also the age threshold is currently used to delete data, this is not used to delete the record, only blank certain fields in the row.

It's probably a tiny detail but I wanted the threshold name more descriptive of why it exists :D

**Review by H3ll3m4 — APPROVED** [review]
🚀

**Pull request merged**